src/enclave/enclave_signal.c

```diff
@@ -151,10 +152,11 @@ static uint64_t sgxlkl_enclave_signal_handler(
 #endif
 
         /**
-         * If LKL has not yet been initialised, we cannot handle the
-         * exception and fail instead.
+         * If LKL has not yet been initialised or is terminating (and thus no
+         * longer accepts signals), we cannot handle the exception and fail
+         * instead.
          */
-        if (!lkl_is_running())
+        if (!lkl_is_running() || is_lkl_terminating())
```

Comment:
small thing.

there appears to only be two usages of lkl_is_running.

both are !lkl_is_running ie testing the negation.

Given this change, I think it would read easier if it was changed to

lkl_isnt_running()

Reply:
Hmm, since that function may change with the re-layering anyway, I don't think it's worth changing it now.